products/BellHybrid/apps/common/src/widgets/TimeSetSpinnerVertical.cpp

```diff
@@ -148,44 +202,69 @@ namespace gui
         const auto t = std::localtime(&time);
 
         if (timeFormat == Locale::TimeFormat::FormatTime24H) {
-            hour->set_value(t->tm_hour);
-            minute->set_value(t->tm_min);
+            const auto hours   = std::chrono::hours{t->tm_hour};
+            const auto time24H = date::make24(hours, isPM());
+            auto stringHour = std::to_string(time24H.count());
+            if(t->tm_hour < doubleDigit) {
+                hour_first->setText("0");
+                hour_second->setText(stringHour.substr(0,1));
+            }
+            else {
+                hour_first->setText(stringHour.substr(0,1));
+                hour_second->setText(stringHour.substr(1,1));
+            }
+
+            auto stringMin = std::to_string(t->tm_min);
+            if(t->tm_min < doubleDigit) {
+                minute_first->setText("0");
+                minute_second->setText(stringMin.substr(0,1));
+            }
+            else {
+                minute_first->setText(stringMin.substr(0,1));
+                minute_second->setText(stringMin.substr(1,1));
+            }
```

Comment:
Isnt this part of code duplicated in 12H format handling? Maybe we can take it out of this conditional and use once?

Reply:
Yes, refactored this part to clean the code